src/Simulation/CSharpGeneration/EntryPoint.fs

```diff
@@ -119,65 +97,169 @@ let private callableTypeNames context (callable : QsCallable) =
+                param simulatorParamName ``of`` (``type`` "string")
+            ]
+        ``)``
+        [``public``]
+        (Some (``=>`` (simulationType <.> (ident "Simulate", args))))
 
 /// The class that adapts the entry point for use with the command-line parsing library and the driver.
-let private entryPointClass context entryPoint =
-    let callableName, argTypeName, returnTypeName = callableTypeNames context entryPoint
+let private entryPointClass context (entryPoint : QsCallable) =
     let property name typeName value = ``property-arrow_get`` typeName name [``public``] get (``=>`` value)
+    let nameProperty =
+        entryPoint.FullName.ToString()
+        |> literal
+        |> property "Name" "string"
     let summaryProperty =
         (PrintSummary entryPoint.Documentation false).Trim ()
         |> literal
         |> property "Summary" "string"
     let parameters = parameters context entryPoint.Documentation entryPoint.ArgumentTuple
-    let defaultSimulator =
-        context.assemblyConstants.TryGetValue AssemblyConstants.DefaultSimulator
-        |> fun (_, value) -> if String.IsNullOrWhiteSpace value then AssemblyConstants.QuantumSimulator else value
-    let defaultSimulatorNameProperty = literal defaultSimulator |> property "DefaultSimulatorName" "string"
-    let defaultExecutionTargetProperty =
-        context.assemblyConstants.TryGetValue AssemblyConstants.ExecutionTarget
-        |> (fun (_, value) -> if value = null then "" else value)
-        |> literal
-        |> property "DefaultExecutionTarget" "string"
-    let infoProperty =
-        property "Info" (sprintf "EntryPointInfo<%s, %s>" argTypeName returnTypeName)
-                        (ident callableName <|.|> ident "Info")
     let members : MemberDeclarationSyntax list = [
+        nameProperty
         summaryProperty
         parameterOptionsProperty parameters
-        defaultSimulatorNameProperty
-        defaultExecutionTargetProperty
-        infoProperty
-        customSimulatorFactory defaultSimulator
         createArgument context entryPoint
-        mainMethod context entryPoint
+        submitMethod context entryPoint
+        simulateMethod context entryPoint
     ]
-    let baseName = sprintf "%s.IEntryPoint<%s, %s>" driverNamespace argTypeName returnTypeName
-    ``class`` entryPointClassName``<<`` [] ``>>``
+    let baseName = sprintf "%s.IEntryPoint" driverNamespace
+    ``class`` ((entryPointClassFullName entryPoint).Name) ``<<`` [] ``>>``
         ``:`` (Some (simpleBase baseName)) ``,`` []
         [``internal``]
         ``{``
             members
         ``}``
 
-/// Generates C# source code for a standalone executable that runs the Q# entry point.
-let generate context (entryPoint : QsCallable) =
-    let ns =
-        ``namespace`` entryPoint.FullName.Namespace
+/// Generates a namespace for a set of entry points that share the namespace
```

Comment:
Is the namespace name dependent on the namespace of the entry point operation, such that potentially multiple such namespaces are created?

Reply:
Yes, there may be more than one namespace with entry point classes in them.